components/molecules/Table.jsx

```diff
@@ -229,7 +350,7 @@ Table.propTypes = {
                             componentName +
                             '`, expected `object`.'
                     );
-                } else if (props.customRowKey && !props.data[i][props.customRowKey]) {
+                } else if (props.customRowKey && typeof props.data[i][props.customRowKey] === 'undefined') {
                     return new Error(
```

Comment:
I changed this logic as this was a loose check, the test data in storybook generates id's as numbers, when the customRowKey points to id which is a unique number, if the value is 0 it will throw the error.